invenio_app_rdm/requests_ui/views/requests.py

```diff
@@ -84,28 +94,40 @@ def _resolve_topic_record(request):
     return dict(permissions={}, record_ui=None)
 
 
-def _resolve_record_or_draft_files(record):
+def _resolve_record_or_draft_files(record, request):
     """Resolve the record's or draft's files."""
+    request_type = request["type"]
+    is_record_inclusion = request_type == CommunityInclusion.type_id
     if record and record["files"]["enabled"]:
         record_pid = record["id"]
         try:
-            files = draft_files_service().list_files(
-                id_=record_pid, identity=g.identity
-            )
+            if is_record_inclusion:
+                files = files_service().list_files(id_=record_pid, identity=g.identity)
+            else:
+                files = draft_files_service().list_files(
+                    id_=record_pid, identity=g.identity
+                )
```

Comment:
minor/shelve: it feels like the logic is a bit duplicated and disconnected here. It should be that based on the `record`'s type, we have a way to fetch the files (without having to depend on the request to determine what the `record` is). In any case, let's shelve and refactor later.